octopusCatsEffect/src/main/scala/octopus/async/cats/cats.scala

```diff
@@ -0,0 +1,25 @@
+package octopus.async
+
+import _root_.cats.effect.IO
+import _root_.cats.implicits._
+import _root_.cats.Applicative
+
+import octopus.AppError
+
+package object cats {
+
+  implicit val catsIOAppError: AppError[IO] = new AppError[IO] {
+    def pure[A](a: A): IO[A] = IO(a)
+
+    def map2[A, B, C](first: IO[A], second: IO[B])(combine: (A, B) => C): IO[C] =
+      Applicative[IO].map2(first, second)(combine)
```

Comment:
note that this is never parallel, so async validations using IO will be run sequentially

Reply:
What about map2 from ApplicativeError? I guess it depends on the instance then?

Reply:
No, it's just Future that will run in parallel with its Applicative instance, because when you call map2 with two futures they will already be both running. With IO/Task and others, you normally need an instance of Parallel and parMap2 etc.

Reply:
Note that there's an Applicative for IO that's parallel, but it's actually for a IO.Par newtype, which is used by Parallel[IO, IO.Par] underneath. We can talk about it later today ;)